src/java.desktop/unix/native/libawt_xawt/awt/awt_GraphicsEnv.c

```diff
@@ -1428,6 +1428,7 @@ Java_sun_awt_X11GraphicsDevice_getDoubleBufferVisuals(JNIEnv *env,
             break;
         }
     }
+    XdbeFreeVisualInfo(visScreenInfo);
```

Comment:
I think we should use AWT_UNLOCK/AWT_UNLOCK here.

Reply:
Fixed. I assume all X11 calls have to be serialized, right?

Thanks.

Reply:
If you mean "and so there's no chance of a deadlock", well no .. but here we only have the one lock in this JNI method and we are using the lock already earlier so it is relatively safe.

Reply:
Thanks for the review, @prrace

Reply:
@mrserb Is it okay now?

Thanks